synthesizer/msynthesizer.cpp

```diff
@@ -408,7 +408,9 @@ bool MasterSynthesizer::storeState()
             }
       XmlWriter xml(0, &f);
       xml.header();
-      state().write(xml);
+      SynthesizerState ss = state();
+      ss.setIsDefault(false);
```

Comment:
This doesn't look like a clean solution. My preference would probably be having a flag like `alwaysWrite` (or maybe name it like `force`) as a parameter to `write()` function which overrides `isDefault()` value:
```
// declaration
void write(XmlWriter& xml, bool alwaysWrite = false) const;
// when storing master synthesizer state
state().write(xml, /* alwaysWrite */ true);
```

Do you agree with that or are there other reasons to make it like it is in this PR?

Reply:
Those would essentially work the same way, I guess, since it all comes down to how its written. We'd have to keep the `_dirty` member of SynthControl though to know whether to set alwaysWrite or not.

Reply:
hang on, looking back over it, that's not really true. We need the `_isDefault` member of SynthState so that we know whether the synth state has been changed in SynthControl. Otherwise there's no way of knowing whether it's been changed from the default state. So I'd say this is the best implementation currently - getting rid of _isDefault may actually make it more convoluted than it is already.

Reply:
No, I don't propose getting rid of `_isDefault` member, I suggest that instead of making a copy of the current synthesizer state and marking it not default just to force it being written you could introduce a boolean flag as a `write()` function parameter which will be used in case the state should be written regardless of `_isDefault` flag state (that is, exactly in this case). That way you could avoid those unnecessary copying and "undefaulting" operations.

Reply:
OK, that sounds like a good idea. I'll do that.

Reply:
Done.